### PR TITLE
Adding a doc for server release process

### PR DESCRIPTION
**Pull request by abhinavg6**
## Description

Add a new doc for server release process, covering the frequency, release notes, downtime aspects, and end of life policy.

## Ticket

NA

## Checklist

Check if your PR fulfills the following requirements. Put an `X` in the boxes that apply.

- [X] Files I edited were previewed on my local development server with `yarn start`. My changes did not break the local preview.
- [ ] Build (`yarn docusaurus build`) was run locally and successfully without errors or warnings. - still getting unrelated TypeError: Invalid URL errors.
- [X] I merged the latest changes from `main` into my feature branch before submitting this PR.

### CODE REVIEW

File: docs/guides/hosting/server-release-process.md [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+* When a infrastructure change is required as part of a security patch or to ensure that we're not hitting `support end-of-life` for a particular version. If such a change needs urgent attention, we may not be able to notify the **Dedicated Cloud** customers in advance. The goal here is to keep the fleet secure & fully supportable at all times.
+
+For both cases above, such updates are applied to the whole **Dedicated Cloud** fleet in batches, and we can not exclude a particular instance from the process. Customers with **Self-managed** instances are responsible to manage such updates on their own schedule. Refer to [release support and end of life](#release-support-and-end-of-life).
+
+## Release support and end of life
+We officially support server releases from last six months. This should not affect the **Dedicated Cloud** instances as we update those as part of the release process. But customers with **Self-managed** instances should ensure that they're updating their deployment(s) with the latest release when possible. Staying on a version older than six months would mean that they have limited to no means of support.
```
Comment (venky-wandb) [inline docs/guides/hosting/server-release-process.md]: > Staying on a version older than six months would mean that they have limited to no means of support.

We may benefit from adding something similar to this into the actual contract itself.

**Review by venky-wandb — APPROVED** [review]
This is great, thank you! 🚀 🚢

File: docs/guides/hosting/server-release-process.md (hunk 1 above)
Comment (abhinavg6) [reply docs/guides/hosting/server-release-process.md]: Yeah, that would be a separate thing later. For now I want to get it into the docs.

**Review by cvphelps — APPROVED** [review]

**Review by ngrayluna — APPROVED** [review]

**Comment by cvphelps** [comment]
Thanks Noah! @abhinavg6 this will go out tomorrow in the docs deployment!

**Pull request merged**